### PR TITLE
Reduce some rebuilding on CI

### PR DESCRIPTION
**Pull request by vkgnosis**
The different crates in our workspace activate different features in dependencies. This means that some dependencies have to be recompiled based on which crate (--package) is being compiled.
This commit avoids this by introducing one global build step first.
Sadly we still recompilations for the e2e tests because the only way to run them is through --package .
Also moved some commands into their own step so they are easier to keep track of in the github actions page output.

### Test Plan
ci still works

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -38,21 +38,21 @@ jobs:
...
           sudo systemctl start postgresql.service
           sudo -u postgres createuser $USER
           sudo -u postgres createdb $USER
           psql -c "ALTER USER $USER PASSWORD '$FLYWAY_PASSWORD';"
-          flyway-${FLYWAY_VERSION}/flyway -url="jdbc:postgresql:///" -user=$USER -locations="filesystem:database/sql/" migrate 
-          # Postgres tests should not run in parallel because they use the same database.
-          cargo test --locked --workspace --all-features --jobs 1 postgres -- --ignored --test-threads 1
+          flyway-${FLYWAY_VERSION}/flyway -url="jdbc:postgresql:///" -user=$USER -locations="filesystem:database/sql/" migrate
+        # Postgres tests should not run in parallel because they use the same database.
+      - run: cargo test --locked --all-features --jobs 1 postgres -- --ignored --test-threads 1
```
Comment (nlordell) [inline .github/workflows/pull-request.yaml:50]:
```suggestion
      - run: cargo test --locked --all-features postgres -- --ignored --test-threads 1
```

Why reduce the number of jobs for building the tests? AFAIK `--jobs` does not affect the number of threads for executing tests.

**Review by nlordell — APPROVED** [review]
Something that may be worth keeping in mind (but I don't think applies here anymore seeing how the `cargo` invocations changed) is a ["workspace hack" crate like the Rust-lang repo uses](https://github.com/rust-lang/rust/tree/master/src/tools/rustc-workspace-hack) in order to deal with different crates needing different features in their shared dependencies?

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (vkgnosis) [reply .github/workflows/pull-request.yaml:50]: I didn't add this in this PR but yeah I think you're right that it isn't needed. Will apply the suggestion.

Comment (nlordell) [reply .github/workflows/pull-request.yaml]: Ah right - didn't notice it was from before.

**Pull request merged**